### DISCUSSION TITLE
One Development and Deployment repo?

### DISCUSSION THREAD

**Discussion by mbvgua** [mkdocs/mkdocs#3856] [Ideas]
Hello. Im currently new to the SSG's but being familiar with python, I opted to got with mkDocs. 

After following the documentation, I realized that I need to have separate repositories for development and deployment with github pages. This is quite counter productive in the long term to me and I was wondering if maybe there was a way to have  both of these in one repository.

I imagine using one repo and having everything on the `main` branch, then utilizing github actions to push any changes you make to the `gh-pages` branch and automatically deploying it, similar to hugo in a way.

**Comment by pawamoy** [comment]
> I realized that I need to have separate repositories for development and deployment with github pages.

That's not correct. Could you point at the docs that lead you to think this?

> I imagine using one repo and having everything on the main branch, then utilizing github actions to push any changes you make to the gh-pages branch and automatically deploying it, similar to hugo in a way.

Yep that's already supported, and that's what the `mkdocs gh-deploy` command does too. Let us know if you have any troubles setting this up :slightly_smiling_face:

**Reply by mbvgua** [reply]
[This](https://www.mkdocs.org/user-guide/deploying-your-docs/) is where I got his from. I am using a user page, hence I have two repos on my local device, one with the main code and another where the website will be contained.

As explained below in the official docs:
![Screenshot from 2024-11-01 17-47-59](https://github.com/user-attachments/assets/1b5ebe16-ece5-4351-86ed-8e1de2b6767e)

Naturally both directories are linked to separate repos on github

**Reply by Andre601** [reply]
You misunderstand things here.
The section you show specifically talks about **User and Organization Pages**. These are special repositories on GitHub that behave slightly different than your normal GitHub Pages repositories.

Their purpose is to allow you to have a page for the main `<username>.github.io` domain itself. To have that, one would make a repository named after the domain (i.e. for me, it would be named `andre601.github.io`).
However, a caveat of it is that - assuming you use the "default" Deployment option on the Repository for GitHub Pages - it uses the default/main branch of your repository to deploy your page, rather than a `gh-pages` branch. That's basically what the section is referring to, as you can't have the source visible on the main branch, requiring it to either be on a different branch, or a different repository altogether.

However, this issue is non-existant, if you use the "GitHub Actions" option for pages deployment, as with that, you deploy directly to GitHub pages using your own actions workflow, instead of haveing GitHub do it for you.

TL;DR: User and Organization Pages refers to a unique type of repository on GitHub that isn't the same as any other repository with a `gh-pages` branch and does not mean it applies to all repositories.

**Reply by mbvgua** [reply]
Ohh. Thanks for the much needed clarity mate! 

I'll be sure to try deployment with github actions soon.
